addons/website/static/src/js/content/menu.js

```diff
@@ -424,9 +424,9 @@ publicWidget.registry.autohideMenu = publicWidget.Widget.extend({
      * @override
      */
     async start() {
-        await this._super(...arguments);
         this.$topMenu = this.$('#top_menu');
         this.noAutohide = this.$el.is('.o_no_autohide_menu');
+        await this._super(...arguments);
```

Comment:
Not a perfect fix for multiple reasons:
- If I look at this code in a year, I will say "Why did someone put that super call in the middle of the function?" and may reintroduce the bug. There should be no need for the super call to be in the middle of the start definition at all in fact.
- There are async parts that can be done before the `start` (like in the `willStart`) so the bug would still be there if the `willStart` does async stuff at some point.
- There is no guarantee that `start` is called at all before `destroy`. `init` could be called then immediately `destroy` and the `destroy` should not crash in that case either.

=> Simpler to understand and more robust fix: just check in the `destroy` method that the variable we use are defined (like we should do in all `destroy` implementation for the same reason).